### PR TITLE
docs: ADR for Ralph as Learning Record Store

### PR DESCRIPTION
**Pull request by bmtcril**

### CODE REVIEW

File: docs/decisions/0005_ralph.rst [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+.. _xapi.com: https://xapi.com/learning-record-store/
+
+Decision
+********
+
+The Open Analytics Reference System (OARS) will use `Ralph`_ as its LRS. This decision is was made for
+several reasons:
+
+- It is a toolkit already rooted in the Open edX ecosystem, and developed by a long-term Open edX partner (OpenFUN)
+- It allows us to us `ClickHouse`_ as a database backend as decided in :doc:`0004_clickhouse`
```
Comment (jmaupetit) [inline docs/decisions/0005_ralph.rst:27]:
```suggestion
- It allows us to use `ClickHouse`_ as a database backend as decided in :doc:`0004_clickhouse`
```

File: docs/decisions/0005_ralph.rst [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+There are many xAPI compatible LRSs, each with their own pluses and minuses, and supporting different
+data storage backends. In discussions with the community while developing the OARS architecture
+no community preference for an LRS was forthcoming.
+
+.. _xapi.com: https://xapi.com/learning-record-store/
+
+Decision
+********
+
+The Open Analytics Reference System (OARS) will use `Ralph`_ as its LRS. This decision is was made for
```
Comment (wilbrdt) [inline docs/decisions/0005_ralph.rst:23]:
```suggestion
The Open Analytics Reference System (OARS) will use `Ralph`_ as its LRS. This decision was made for
```

File: docs/decisions/0005_ralph.rst [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+.. _ClickHouse: https://clickhouse.com/
+.. _Ralph: https://openfun.github.io/ralph/
+
+
+Consequences
+************
+
+Ralph will be integrated into OARS via a Tutor plugin. It will be configured to write xAPI statements to a
+single table, and other OARS components such will manage the downstream reporting schema separately.
```
Comment (wilbrdt) [inline docs/decisions/0005_ralph.rst:41]:
```suggestion
single table, and other OARS components will manage the downstream reporting schema separately.
```

File: docs/decisions/0005_ralph.rst [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+A single Ralph instance should support a medium sized Open edX instance on commodity hardware,
+including writing to ClickHouse while it is under normal reporting load for a small to medium Open edX install.
+Large scale deployments, extremely intensive reports, or novel use cases outside the scope of OARS v1.
```
Comment (wilbrdt) [inline docs/decisions/0005_ralph.rst:45]:
```suggestion
Large scale deployments, extremely intensive reports, or novel use cases are outside the scope of OARS v1.
```

**Review by wilbrdt — COMMENTED** [review]
Looks good!

**Comment by bmtcril** [comment]
@jmaupetit @wilbrdt thanks for the reviews, I've taken all of the feedback and squashed. What do you think?

**Review by wilbrdt — APPROVED** [review]
Perfect!

File: docs/decisions/0005_ralph.rst [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+A single Ralph instance should support a medium sized Open edX instance on commodity hardware,
+including writing to ClickHouse while it is under normal reporting load for a small to medium Open edX install.
+Large scale deployments, extremely intensive reports, or novel use cases are outside the scope of OARS v1.
+
+Ralph does not currently support the entire xAPI LRS specification, only the `statements` endpoint, which is
+all that we require for OARS v1. OpenFUN has been happy to receive pull requests and would welcome new contributions if further endpoints need
+to be developed.
+
+As Ralph's `statements` endpoint is xAPI compliant, other LRSs can be used for xAPI storage, however doing so
+would likely mean losing the other OARS functionality (Superset reporting) if it does not support ClickHouse
```
Comment (sarina) [inline docs/decisions/0005_ralph.rst:52]:
```suggestion
would likely mean losing the other OARS functionality (such as Superset reporting) if it does not support ClickHouse
```

Nit: just to hedge your bets that other functionality may not work

File: docs/decisions/0005_ralph.rst [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+.. _Ralph: https://openfun.github.io/ralph/
+
+
+Consequences
+************
+
+Ralph will be integrated into OARS via a Tutor plugin. It will be configured to write xAPI statements to a
+single table, and other OARS components will manage the downstream reporting schema separately.
+
+A single Ralph instance should support a medium sized Open edX instance on commodity hardware,
```
Comment (sarina) [inline docs/decisions/0005_ralph.rst:43]: Is there a canonical definition of "small" and "medium" sized instance?

**Review by sarina — APPROVED** [review]

**Pull request merged**